sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -579,7 +579,7 @@ class SQLContext private[sql](
     val className = beanClass.getName
     val beanInfo = Introspector.getBeanInfo(beanClass)
     val rows = SQLContext.beansToRows(data.asScala.iterator, beanInfo, attrSeq)
-    DataFrame(self, LocalRelation(attrSeq, rows.toSeq))
+    DataFrame(self, LocalRelation(attrSeq, rows.toArray))
```

Comment:
I think it was the same before -- it was materialized into a `Seq`. I think Josh is suggesting this is inherently necessary before making a `LocalRelation` so there's no meaningful opportunity for lazy eval (?)  If so then this seems fine.